daft/logical/logical_plan.py

```diff
@@ -593,6 +593,29 @@ def rebuild(self) -> LogicalPlan:
         return GlobalLimit(input=self._children()[0].rebuild(), num=self._num)
 
 
+class LocalCount(UnaryNode):
+    def __init__(self, input: LogicalPlan) -> None:
+        schema = Schema([Field("count", ExpressionType.integer())])
+        super().__init__(schema, partition_spec=input.partition_spec(), op_level=OpLevel.PARTITION)
+        self._register_child(input)
+
+    def __repr__(self) -> str:
+        return self._repr_helper()
+
+    def copy_with_new_children(self, new_children: list[LogicalPlan]) -> LogicalPlan:
+        assert len(new_children) == 1
+        return LocalCount(new_children[0])
+
+    def required_columns(self) -> set[str]:
+        return set()
```

Comment:
This is technically true, but actually might mess with the way our optimizer works. Our optimizer might see this and think "none of the columns are required, so no work needs to be done".

Instead, let's do the naive thing and just grab the first column from the child plan for now?

We can add a comment here explaining why this is done.

```suggestion
        # HACK: Arbitrarily return the first column in the child to ensure that
        # at least one column is computed by the optimizer
        return self._children()[0].schema().column_names()[0]
```